### PR TITLE
Fix flaky session tests

### PR DESCRIPTION
**Pull request by lucas-clemente**

### CODE REVIEW

File: session_test.go [hunk 1]
```diff
@@ -1179,7 +1179,7 @@ var _ = Describe("Session", func() {
 			sess.ackAlarmChanged(time.Now().Add(10 * time.Millisecond))
 			time.Sleep(10 * time.Millisecond)
 			Eventually(func() int { return len(mconn.written) }).ShouldNot(BeZero())
-			Expect(mconn.written[0]).To(ContainSubstring(string([]byte{0x37, 0x13})))
+			Eventually(func() []byte { return mconn.written[0] }).Should(ContainSubstring("foobar"))
```
Comment (marten-seemann) [inline session_test.go]: I don't really understand what we're testing here, and why it is valid to not test for the ACK anymore. Can we find a more meaningful name for the test?

File: session_test.go [hunk 2]
```diff
@@ -1198,8 +1198,8 @@ var _ = Describe("Session", func() {
 				defer sess.Close(nil)
 
 				Eventually(func() [][]byte { return mconn.written }).Should(HaveLen(1))
-				Expect(mconn.written[0]).To(ContainSubstring("foobar1"))
-				Expect(mconn.written[0]).To(ContainSubstring("foobar2"))
+				Eventually(func() []byte { return mconn.written[0] }).Should(ContainSubstring("foobar"))
+				Eventually(func() []byte { return mconn.written[0] }).Should(ContainSubstring("foobar"))
```
Comment (marten-seemann) [inline session_test.go]: You should test for foobar1 and foobar2 here.

File: session_test.go (hunk 1 above)
Comment (lucas-clemente) [inline session_test.go]: Copy & paste error, fix pushed.

File: session_test.go (hunk 2 above)
Comment (lucas-clemente) [inline session_test.go]: Same as above.

**Comment by lucas-clemente** [comment]
The two commits fix issues with different things (a sessino member vs the mock conn), so I'd like to keep them separate.

**Comment by marten-seemann** [comment]
Oh, I didn't see that. Why does Github hide the second line of the commit message here?

**Comment by lucas-clemente** [comment]
It's github. They don't do UX.

**Comment by lucas-clemente** [comment]
This seems to make matters worse the more races I fix. I don't think this approach here will get us anywhere soon, and I don't have the time right now to massively rewrite the tests.

**Pull request closed without merging**